### PR TITLE
IN-489 run a db validation level check after migration - Clients

### PR DESCRIPTION
**Pull request by jackgoodby**
## Purpose

First in a series of assurance scripts which will allow us to quickly check that the data has made it all the way from csv to Sirius prod - column by column

## Approach

First the casrecmigration.etl1 schema is copied to sirius.casrec_csv
The same set of data with the same ordering is selected from sirius.casrec_csv and sirius.api (prod db)
Any differences (exceptions) are inserted as a row into a corresponding exception table
At the end, the number of rows in the exception tables are counted (hopefully zero) and displayed

The data on the casrec side (the csv side) is all text, so a lot of the complexity with this is applying the correct data transformations to compare like for like

The data in casrec is also often subject to transformation, so similar transformations are applied to that data

## Learning

Much of this methodology lifted from my own work on the Meris migration in Aug 2019, and changed quite a bit to suit our needs

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [x] I have updated documentation where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: migration_steps/validation/validate_db/app/app.py [hunk 1]
```diff
@@ -96,11 +69,178 @@ def main(verbose):
...
+        "client_phonenumbers": "phonenumbers"
+    }
+    return mapping_name_to_table.get(mapping, mapping)
+
+
+def get_exception_table(mapping):
+    return f"casrec_migration_exceptions_{mapping}"
+
+
+def get_map_dict(mapping):
```
Comment (jmackown) [inline migration_steps/validation/validate_db/app/app.py:85]: Very similar method in `migration_steps/shared/helpers.py` - could you use/modify the shared verison instead?

File: migration_steps/validation/validate_db/app/app.py [hunk 2]
```diff
@@ -96,11 +69,178 @@ def main(verbose):
...
+    sql_file.writelines(sql)
+    sql_file.close()
+
+
+def show_results(conn_target):
+    build_results_sql()
+    results_df = df_from_sql_file(
+        sql_path, "get_validation_results.sql", conn_target
+    )
+    print(
```
Comment (jmackown) [inline migration_steps/validation/validate_db/app/app.py:200]: Can you swap `print` statements for logs? Gives us more control over what data appears where.

**Review by jmackown — DISMISSED** [review]

File: migration_steps/validation/validate_db/app/app.py (hunk 1 above)
Comment (jackgoodby) [reply migration_steps/validation/validate_db/app/app.py]: Done :)

File: migration_steps/validation/validate_db/app/app.py (hunk 2 above)
Comment (jackgoodby) [reply migration_steps/validation/validate_db/app/app.py]: Done

**Review by jamesrwarren — APPROVED** [review]

**Pull request merged**